File: task-03/Go/code3..go
```go
package main

import (
    "fmt"
)

func printDiamond(n int) {
    for i := 0; i < n; i++ {
        fmt.Printf("%s%s\n", string(' ', n-i-1), string('*', 2*i+1))
    }
    for i := n - 2; i >= 0; i-- {
        fmt.Printf("%s%s\n", string(' ', n-i-1), string('*', 2*i+1))
    }
}

func main() {
    var n int
    fmt.Print("Enter a number: ")
    fmt.Scan(&n)
    printDiamond(n)
}
```

File: task-03/Haskell/code3..hs
```haskell
import Control.Monad (replicateM_)

printDiamond :: Int -> IO ()
printDiamond n = do
    replicateM_ n $ \i -> do
        putStrLn $ replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*'
    replicateM_ (n - 2) $ \i -> do
        putStrLn $ replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*'

main :: IO ()
main = do
    putStr "Enter a number: "
    n <- readLn
    printDiamond n
```
